frontend/views/components/Graphs/Bars.vue

```diff
@@ -1,48 +1,223 @@
+      const scales = []
+      let scale = Math.ceil(this.min / roundedTickRange) * roundedTickRange
+      while (scale <= this.max) {
+        let label = this.currency(Math.abs(scale))
+        if (scale < 0) label = '-' + label
+        // Add scale label and positition
+        scales.push({ label: label, position: this.calculRatioY(scale) })
+        scale = scale + roundedTickRange
+      }
+      return scales
+    }
   },
   methods: {
-    getOpacity (percentage) {
-      return percentage > 1 ? '' : percentage
+    handleResize: debounce(function () {
+      if (this.$refs.graph) {
+        this.isMobile = this.verifyIsMobile()
+        this.ratioX = this.$refs.graph.clientWidth
+        this.availableWidth = this.ratioX - this.labelwidth
+      }
+    }, 100),
+
+    toPrecision (nbr) {
+      if (typeof nbr !== 'number') return 0
+      if (nbr === 0) return 0
+      // log only for positive number
+      const num = Math.abs(nbr)
+      // Remove floating number for large number to calcul the precision
+      if (nbr > 10) nbr = parseInt(nbr)
+      // Increase precision for small value
+      const precision = nbr.toString().length > 3 ? 2 : 1
+      // Rounding technic
+      const digits = Math.ceil(Math.log(num) / Math.LN10)
+      const factor = Math.pow(10, precision - digits)
+      let result = Math.round(num * factor, 0) / factor
+      // Remove floating number for large number
+      if (num > 1000) result = parseInt(result)
+      // Bring sign back
+      return nbr > 0 ? result : -result
+    },
+
+    verifyIsMobile () {
+      return window.innerWidth < TABLET
+    },
+
+    calculRatioY (y) {
+      return this.ratioY / (this.max + Math.abs(this.min)) * y
+    },
+
+    height (delta) {
+      return this.calculRatioY(Math.abs(delta))
+    },
+
+    positionX (index) {
+      const marginLeft = Math.min((this.availableWidth - (this.width * this.membersNumber)) / this.membersNumber, 28)
+      const maxWidth = this.membersNumber * (this.width + marginLeft)
+      const positionX = maxWidth / this.membersNumber * index
+      const offset = (this.availableWidth - maxWidth) / 2 - marginLeft
+      return `translate(${positionX + marginLeft + offset}, 0)`
     },
-    getHeight (percentage) {
-      return percentage > 1.25 ? 1.25 : percentage
+
+    positionY (delta) {
+      // If delta is positive, the bar start at 0 otherw
+      return delta >= 0 ? -(this.calculRatioY(delta)) : 0
+    },
+
+    color (delta, surplus) {
+      if (delta >= 0) return surplus ? 'g-positive' : 'g-surplus'
+      else return surplus ? 'g-negative' : 'g-needed'
     },
-    getTotal () {
-      return '$4500'
+
+    roundedRect (up, x, y, w, h, r) {
+      let retval = 'M' + (x + r) + ',' + y
+      retval += 'h' + (w - 2 * r)
+      // Top right corner
+      if (up) retval += 'a' + r + ',' + r + ' 0 0 1 ' + r + ',' + r // Arc
+      else {
+        // Square corner
+        retval += 'h' + r
+        retval += 'v' + r
+      }
```

Comment:
Following @mmbotelho designs, the blue shouldn't be rounded when there's green as well: (left vs right = implemented vs designed).

![image](https://user-images.githubusercontent.com/14869087/72131869-63656100-3375-11ea-9233-0886a477ebc3.png)